src/execution_space_provider/utilities/external_provider.py

```
@@ -202,24 +209,24 @@ def start(self, minimum_amount: int, maximum_amount: int) -> str:
                 "RABBITMQ_PORT": str(rabbitmq.get("port")),
                 "RABBITMQ_VHOST": rabbitmq.get("vhost"),
                 "RABBITMQ_SSL": str(rabbitmq.get("ssl")).lower(),
-                "SOURCE_HOST": self.etos.config.get("source").get("host"),
+                "SOURCE_HOST": source.get("host"),
                 "ETOS_GRAPHQL_SERVER": self.etos.debug.graphql_server,
                 "ETOS_API": self.etos.debug.etos_api,
                 "ETR_VERSION": os.getenv(
                     "ETR_VERSION",
                 ),
             },
             "artifact_id": self.dataset.get("artifact_id"),
-            "artifact_created": self.dataset.get("artifact_created"),
-            "artifact_published": self.dataset.get("artifact_published"),
-            "tercc": self.dataset.get("tercc"),
+            "artifact_created": self.dataset.get("artifact_created") or {},
```

Are these fields allowed to be empty dictionaries? 
Is some additional validation needed here or anywhere else?

---

They are allowed to be empty and null. Validation is expected in the external provider, but it shall not fail if these are not set.